From 2b95f677bb434c4aa412358fba0f2c04cb00f892 Mon Sep 17 00:00:00 2001
From: Kuan-Chih Wang <kuanchihwang@outlook.com>
Date: Sat, 3 Jan 2026 20:54:04 +0800
Subject: [PATCH 2/3] Fix feature detection logic for netCDF-4 capability

As of netCDF-C version 4.9.3, NC_HAS_NC4 is no longer defined.
netCDF-4 capability is now detected through NC_HAS_HDF5.

The original logic, which relied on NC_VERSION_PATCH, was also
flawed anyway because it did not check NC_VERSION_{MAJOR,MINOR}.
---
 CMakeLists.txt          | 2 +-
 configure.ac            | 2 +-
 src/clib/pioc_support.c | 5 +++--
 3 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7cb025be..7d360f43 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -321,7 +321,7 @@ SET(STATUS_PNETCDF ${PnetCDF_C_FOUND})
 ###
 CHECK_C_SOURCE_COMPILES("
 #include <netcdf_meta.h>
-#if !NC_HAS_NC4
+#if !NC_HAS_HDF5
       choke me
 #endif
 int main() {return 0;}" HAVE_NETCDF4)
diff --git a/configure.ac b/configure.ac
index d44688da..5207a6a2 100644
--- a/configure.ac
+++ b/configure.ac
@@ -261,7 +261,7 @@ fi
 
 # Do we have netCDF-4?
 AC_COMPILE_IFELSE([AC_LANG_PROGRAM([#include "netcdf_meta.h"],
-[[#if !NC_HAS_NC4
+[[#if !NC_HAS_HDF5
 # error
 #endif]
 ])], [have_netcdf4=yes], [have_netcdf4=no])
diff --git a/src/clib/pioc_support.c b/src/clib/pioc_support.c
index c9139196..ff28b030 100644
--- a/src/clib/pioc_support.c
+++ b/src/clib/pioc_support.c
@@ -3126,8 +3126,9 @@ iotype_is_valid(int iotype)
         ret++;
 
     /* Some builds include netCDF-4. */
-    /* as of netcdf 4.9.3 NC_HAS_NC4 is no longer defined */
-#if NC_HAS_NC4 || (NC_VERSION_PATCH > 2)
+    /* As of netCDF-C version 4.9.3, NC_HAS_NC4 is no longer defined.
+       netCDF-4 capability is now detected through NC_HAS_HDF5. */
+#if NC_HAS_HDF5
     if (iotype == PIO_IOTYPE_NETCDF4C || iotype == PIO_IOTYPE_NETCDF4P)
         ret++;
 #endif /* _NETCDF4 */
-- 
2.47.3

